pkg/install/deploystorage.go

```diff
@@ -103,12 +104,29 @@ func (i *Installer) deployStorageTemplate(ctx context.Context, installConfig *in
 
 		applications := graphrbac.NewApplicationsClient(spp.TenantID, spGraphAuthorizer)
 
-		res, err := applications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
+		timeoutCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+		defer cancel()
+		// NOTE: Do not override err with the error returned by wait.PollImmediateUntil.
+		// Doing this will not propagate the latest error to the user in case when wait exceeds the timeout
+		wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+			var res azgraphrbac.ServicePrincipalObjectResult
+			res, err = applications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
+			if err != nil {
+				i.log.Info(err)
```

Comment:
nit- this could move under strings.Contains() because if it's not an Authorization_IdentityNotFound we will already log this in reply()